operator/ingress/v1/doc.go

```diff
@@ -0,0 +1,7 @@
+// +k8s:deepcopy-gen=package,register
+// +k8s:defaulter-gen=TypeMeta
+// +k8s:openapi-gen=true
+
+// +kubebuilder:validation:Optional
```

Comment:
btw because you have this in your doc.go, it sets every field in the package to optional by default, so you don't need any of the `+kubebuilder:validation:Optional` or `+optional` comments for CRD generation. Not a big deal but just letting you know

Reply:
Yeah — I actually would almost rather explicitly mark every single field with optional or required. The mostly invisible defaulting from the package doc file seems like a recipe for confusion and errors